src/api-client.ts

```diff
@@ -0,0 +1,114 @@
+import * as httpm from "@actions/http-client";
+
+const userAgent = "GitHub multi-repository variant analysis";
+
+export function getApiClient() {
+  return new httpm.HttpClient(userAgent, [], {
+    allowRetries: true,
+  });
+}
+
+interface InProgressProperties {
+  status: "in_progress";
+}
+
+interface SuccessProperties {
+  status: "succeeded";
+  sourceLocationPrefix: string;
+  resultCount: number;
+  databaseSHA: string;
+}
+
+interface FailureProperties {
+  status: "failed";
+  failureMessage: string;
+}
+
+type updateVariantAnalysisProperties =
```

Comment:
Lint police :D Types should start with capital letter.

Reply:
Not in this PR as it might change a lot of things, but shall we enable https://typescript-eslint.io/rules/naming-convention and then it'll be linted for us?

Reply:
For the record, if you turn it on it finds this:
```
/Users/robertbrignull/github/codeql-variant-analysis-action/src/api-client.ts
  27:6  error  Type Alias name `updateVariantAnalysisProperties` must match one of the following formats: PascalCase  @typescript-eslint/naming-convention
  90:3  error  Parameter name `artifact_size` must match one of the following formats: camelCase                      @typescript-eslint/naming-convention
  94:5  error  Object Literal Property name `content_type` must match one of the following formats: camelCase         @typescript-eslint/naming-convention

/Users/robertbrignull/github/codeql-variant-analysis-action/src/interpret.ts
  14:3   error  Type Property name `results_count` must match one of the following formats: camelCase                     @typescript-eslint/naming-convention
  15:3   error  Type Property name `bqrs_file_size` must match one of the following formats: camelCase                    @typescript-eslint/naming-convention
  16:3   error  Type Property name `sarif_file_size` must match one of the following formats: camelCase                   @typescript-eslint/naming-convention
  17:3   error  Type Property name `source_location_prefix` must match one of the following formats: camelCase            @typescript-eslint/naming-convention
  50:11  error  Variable name `bqrs_file_size` must match one of the following formats: camelCase, UPPER_CASE             @typescript-eslint/naming-convention
  53:9   error  Variable name `sarif_file_size` must match one of the following formats: camelCase, UPPER_CASE            @typescript-eslint/naming-convention
  63:7   error  Object Literal Property name `results_count` must match one of the following formats: camelCase           @typescript-eslint/naming-convention
  64:7   error  Object Literal Property name `bqrs_file_size` must match one of the following formats: camelCase          @typescript-eslint/naming-convention
  65:7   error  Object Literal Property name `sarif_file_size` must match one of the following formats: camelCase         @typescript-eslint/naming-convention
  66:7   error  Object Literal Property name `source_location_prefix` must match one of the following formats: camelCase  @typescript-eslint/naming-convention

✖ 13 problems (13 errors, 0 warnings)
```

So most (but not all) of those are property names that will be serialized, so we can't change them super easily and might need to exclude all those from the check. But overall would still be useful to have this checked automatically.

Reply:
Yes please!